lib/ExternalModuleFactoryPlugin.js

```diff
@@ -298,5 +306,42 @@ class ExternalModuleFactoryPlugin {
 			}
 		);
 	}
+
+	/**
+	 * @param {ExternalsType} externalType default external type
+	 * @param {ModuleDependency} dependency module dependency
+	 * @param {DependencyType} dependencyType dependency category
+	 * @returns {ExternalsTypeLiteral} literal external type
+	 */
+	resolveGlobalType(externalType, dependency, dependencyType) {
+		if (typeof externalType === "string") {
+			return externalType;
+		}
+
+		/** @type {ExternalsCategory} */
+		const esmType =
+			dependency instanceof HarmonyImportDependency
+				? "static-import"
+				: dependency instanceof ImportDependency
+					? "dynamic-import"
+					: undefined;
+
+		/** @type {ExternalsCategory} */
+		const externalCategory =
+			dependencyType === "esm"
+				? esmType
+				: /** @type {ExternalsCategory} */ (dependencyType);
+
+		const resolved = externalType[externalCategory] || externalType.fallback;
```

Comment:
Should we have a default value for `fallback`?

Reply:
I think it's hard to decide for user which type should be used as fallback, and it might introduce more implicit settings.